code/modules/mob/living/basic/lavaland/raptor/raptor_ai.dm

```diff
@@ -0,0 +1,180 @@
+
+/datum/ai_planning_subtree/find_and_hunt_target/heal_raptors/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+    if(!controller.blackboard[BB_BASIC_MOB_HEALER])
+        return
+    return ..()
+
+/datum/ai_planning_subtree/find_nearest_thing_which_attacked_me_to_flee/raptor
+    target_key = BB_BASIC_MOB_FLEE_TARGET
+
+/datum/ai_planning_subtree/find_nearest_thing_which_attacked_me_to_flee/raptor/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+    if(!controller.blackboard[BB_RAPTOR_COWARD])
+        return
+    return ..()
+
+/datum/ai_planning_subtree/find_and_hunt_target/raptor_start_trouble
+	target_key = BB_RAPTOR_VICTIM
+	hunting_behavior = /datum/ai_behavior/hunt_target/unarmed_attack_target/bully_raptors
+	finding_behavior = /datum/ai_behavior/find_hunt_target/raptor_victim
+	hunt_targets = list(/mob/living/basic/mining/raptor)
+	hunt_chance = 30
+	hunt_range = 9
+
+/datum/ai_behavior/find_hunt_target/raptor_victim
+
+/datum/ai_behavior/find_hunt_target/raptor_victim/valid_dinner(mob/living/source, mob/living/target, radius)
+    if(target.ai_controller?.blackboard[BB_RAPTOR_TROUBLE_MAKER])
+        return FALSE
+    return can_see(source, target, radius) && target.stat != DEAD
+
+/datum/ai_behavior/hunt_target/unarmed_attack_target/bully_raptors
+    always_reset_target = TRUE
+
+/datum/ai_behavior/hunt_target/unarmed_attack_target/bully_raptors/finish_action(datum/ai_controller/controller, succeeded, hunting_target_key, hunting_cooldown_key)
+    if(succeeded)
+        controller.blackboard[BB_RAPTOR_TROUBLE_COOLDOWN] = world.time + 30 SECONDS
+    return ..()
+
+/datum/ai_planning_subtree/find_and_hunt_target/raptor_start_trouble/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+    if(controller.blackboard[BB_BASIC_MOB_HEALER] || !controller.blackboard[BB_RAPTOR_TROUBLE_MAKER])
+        return
+    if(world.time < controller.blackboard[BB_RAPTOR_TROUBLE_COOLDOWN])
+        return
+    return ..()
+
+/datum/ai_planning_subtree/find_and_hunt_target/care_for_young
+	target_key = BB_RAPTOR_BABY
+	hunting_behavior = /datum/ai_behavior/hunt_target/care_for_young
+	finding_behavior = /datum/ai_behavior/find_hunt_target/raptor_baby
+	hunt_targets = list(/mob/living/basic/mining/raptor/baby_raptor)
+	hunt_chance = 75
+	hunt_range = 9
+
+/datum/ai_planning_subtree/find_and_hunt_target/care_for_young/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+    if(!controller.blackboard[BB_RAPTOR_MOTHERLY])
+        return
+    return ..()
+
+/datum/ai_behavior/find_hunt_target/raptor_baby/valid_dinner(mob/living/source, mob/living/target, radius)
+	return can_see(source, target, radius) && target.stat != DEAD
+
+/datum/ai_behavior/hunt_target/care_for_young
+    always_reset_target = TRUE
+
+/datum/ai_behavior/hunt_target/care_for_young/target_caught(mob/living/hunter, atom/hunted)
+    hunter.manual_emote("grooms [hunted]!")
```

Comment:
I could be wrong about this but shouldn't this `manual_emote` be handled wherever grooming is handled?

Reply:
Because if so this behavior can be replaced with a generic "hunt and click on with combat mode off" behavior